core/src/main/java/org/keycloak/representations/admin/v2/ClientRepresentation.java

```diff
@@ -0,0 +1,246 @@
+        this.webOrigins = webOrigins;
+    }
+
+    public Set<String> getRoles() {
+        return roles;
+    }
+
+    public void setRoles(Set<String> roles) {
+        this.roles = roles;
+    }
+
+    public ServiceAccount getServiceAccount() {
+        return serviceAccount;
+    }
+
+    public void setServiceAccount(ServiceAccount serviceAccount) {
+        this.serviceAccount = serviceAccount;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_ABSENT)
+    public static class Auth {
+
+        @JsonPropertyDescription("Whether authentication is enabled for this client")
+        private Boolean enabled;
+
+        @JsonPropertyDescription("Which authentication method is used for this client")
+        private String method;
+
+        @JsonPropertyDescription("Secret used to authenticate this client with Secret authentication")
+        private String secret;
+
+        @JsonPropertyDescription("Public key used to authenticate this client with Signed JWT authentication")
+        private String certificate;
+
+        public Boolean getEnabled() {
+            return enabled;
+        }
+
+        public void setEnabled(Boolean enabled) {
+            this.enabled = enabled;
+        }
+
+        public String getMethod() {
+            return method;
+        }
+
+        public void setMethod(String method) {
+            this.method = method;
+        }
+
+        public String getSecret() {
+            return secret;
+        }
+
+        public void setSecret(String secret) {
+            this.secret = secret;
+        }
+
+        public String getCertificate() {
+            return certificate;
+        }
+
+        public void setCertificate(String certificate) {
+            this.certificate = certificate;
+        }
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_ABSENT)
+    public static class ServiceAccount {
+
+        @JsonPropertyDescription("Whether the service account is enabled")
+        private Boolean enabled;
+
+        @JsonInclude(JsonInclude.Include.NON_EMPTY)
+        @JsonPropertyDescription("Roles assigned to the service account")
+        private Set<String> roles = new LinkedHashSet<String>();
+
+        public Boolean getEnabled() {
+            return enabled;
+        }
+
+        public void setEnabled(Boolean enabled) {
+            this.enabled = enabled;
+        }
+
+        public Set<String> getRoles() {
+            return roles;
+        }
+
+        public void setRoles(Set<String> roles) {
+            this.roles = roles;
+        }
+    }
+
+    public static void main(String[] args) throws JsonProcessingException {
+        ClientRepresentation rep = new ClientRepresentation();
+        rep.setAuth(new Auth());
+        ObjectMapper mapper = new ObjectMapper();
+        var reader = mapper.readerForUpdating(rep);
+        ClientRepresentation patch = new ClientRepresentation();
+        patch.getAdditionalFields().put("auth", null);
```

Comment:
```suggestion
        patch.removeAdditionalField("auth");
```

Would it be possible to have sth like this?